### PR TITLE
test: Remove useless tests

### PR DESCRIPTION
**Pull request by HowJMay**
Function 'mm_shufflehi_epi16_function' and
'mm_shufflelo_epi16_function' are tool functions for
`_mm_shufflehi_epi16` and `_mm_shufflelo_epi16`, respectively.

So we can removed them from test cases.

### CODE REVIEW

File: tests/impl.h [hunk 1]
```diff
@@ -7,532 +7,530 @@
     case it_##c:                  \
         ret = test_##c(*this, i); \
         break;
-#define INTRIN_FOREACH(TYPE)          \
```
Comment (jserv) [inline tests/impl.h:10]: Why did you change many lines instead of two?

Comment (HowJMay) [reply tests/impl.h:10]: It is caused by the formatter

**Review by marktwtn — APPROVED** [review]
